### PR TITLE
Fixup rpimem to cleanly unload and reload, and make checkpatch clean.

### PR DESCRIPTION
**Pull request by 6by9**
Issue noted when JC was trying to unload and reload an overlay for the V4L2 HEVC codec.
Made checkpatch clean (other than DT compatible string) whilst I was at it.

### CODE REVIEW

File: drivers/char/broadcom/rpivid-mem.c [hunk 1]
```diff
@@ -197,22 +197,29 @@ static int rpivid_mem_probe(struct platform_device *pdev)
...
 		oldname[4] = 'o';
 		oldname[5] = 'n';
-		(void)device_create(priv->class, NULL, priv->devid + 1, NULL,
-				       oldname + 1);
+		old_mem_dev = device_create(priv->class, NULL, priv->devid + 1,
+					    NULL, oldname + 1);
 		kfree(oldname);
+
+		ptr_err = old_mem_dev;
+		if (IS_ERR(old_mem_dev))
```
Comment (pelwell) [inline drivers/char/broadcom/rpivid-mem.c]: IS_ERR(ptr_err) would seem more logical here. Or just drop the old_mem_dev and assign directly to ptr_err.

File: drivers/char/broadcom/rpivid-mem.c [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// SPDX-License-Identifier: GPL-2.0-only
```
Comment (pelwell) [inline drivers/char/broadcom/rpivid-mem.c]: The SPDX declaration doesn't appear to match the licence text, which looks like a dual 3-clause BSD and GPL2.

File: drivers/char/broadcom/rpivid-mem.c (hunk 1 above)
Comment (6by9) [reply drivers/char/broadcom/rpivid-mem.c]: I'd copied and pasted from the other device_create failure case. Having ptr_err seems redundant anyway, and the error case may as well do ```err = PTR_ERR(rpivid_mem_dev)``` or ```err = PTR_ERR(rpivid_mem_dev)```.

Actually the handling of ```dev``` as being ```&pdev->dev```, and is then assigned to ```priv-dev``` leaves ```dev``` unused through the rest of the function. I'll do a few more lines of cleanup.

File: drivers/char/broadcom/rpivid-mem.c (hunk 2 above)
Comment (6by9) [reply drivers/char/broadcom/rpivid-mem.c]: Oops, missed that!

**Comment by 6by9** [comment]
Updated. Reworked the conversion from PTR to ERR overall. 
Added an extra patch to remove a deref of ```priv``` when we've just freed it too!

**Pull request merged**